v2/client/idtoken.go

```diff
@@ -1,25 +1,22 @@
-package main
+package client
```

Comment:
A nitpick, but it might be possible to confuse Google's idtoken library used by `WithIDTokenAuth` with this file. Maybe it should be called `default_idtoken`/`default_token` to match it with `WithDefaultTokenAuth`?